src/backends/github/API.js

```diff
@@ -20,70 +21,47 @@ export default class API {
   }
 
   isCollaborator(user) {
-    return this.request('/user/repos').then((repos) => {
-      let contributor = false
-      for (const repo of repos) {
-        if (repo.full_name === this.repo && repo.permissions.push) contributor = true;
-      }
-      return contributor;
-    }).catch((error) => {
-      console.error("Problem with response of /user/repos from GitHub");
-      throw error;
-    })
+    return this.request('/user/repos').then(repos =>
+      Object.keys(repos).some(
+        key => (repos[key].full_name === this.repo) && repos[key].permissions.push
+      )
+    );
   }
 
   requestHeaders(headers = {}) {
-    const baseHeader = {
-      "Content-Type": "application/json",
+    return {
       ...headers,
+      ...(this.token ? { Authorization: `token ${ this.token }` } : {}),
+      "Content-Type": "application/json",
     };
-
-    if (this.token) {
-      baseHeader.Authorization = `token ${ this.token }`;
-      return baseHeader;
-    }
-
-    return baseHeader;
-  }
-
-  parseJsonResponse(response) {
-    return response.json().then((json) => {
-      if (!response.ok) {
-        return Promise.reject(json);
-      }
-
-      return json;
-    });
   }
 
   urlFor(path, options) {
-    const cacheBuster = new Date().getTime();
-    const params = [`ts=${cacheBuster}`];
-    if (options.params) {
-      for (const key in options.params) {
-        params.push(`${ key }=${ encodeURIComponent(options.params[key]) }`);
-      }
-    }
-    if (params.length) {
-      path += `?${ params.join("&") }`;
-    }
-    return this.api_root + path;
+    const cacheBuster = `ts=${ new Date().getTime() }`;
+    const encodedParams = options.params
+          ? Object.entries(options.params).map(
+            ([key, val]) => `${ key }=${ encodeURIComponent(val) }`)
+          : [];
+    return `${ this.api_root }${ path }?${ [cacheBuster, ...encodedParams].join("&") }`;
   }
 
   request(path, options = {}) {
     const headers = this.requestHeaders(options.headers || {});
     const url = this.urlFor(path, options);
-    let responseStatus;
-    return fetch(url, { ...options, headers }).then((response) => {
-      responseStatus = response.status;
+    return fetch(url, { ...options, headers })
+    .then((response) => {
       const contentType = response.headers.get("Content-Type");
       if (contentType && contentType.match(/json/)) {
-        return this.parseJsonResponse(response);
+        return Promise.all([response, response.json()]);
       }
-      return response.text();
+      return Promise.all([response, response.text()]);
     })
-    .catch((error) => {
-      throw new APIError(error.message, responseStatus, 'GitHub');
+    .catch(err => [err, null])
```

Comment:
@Benaiah [You mentioned that](https://github.com/netlify/netlify-cms/pull/517#discussion_r135940695):
> This should be `.catch(err => Promise.reject([err, null))` so the other `.catch` block catches and destructures it properly. This is my bad - it's the same way in my GH API refactor PR.

Reply:
That's correct, thanks!